dcicutils/deployment_utils.py

```diff
@@ -287,11 +312,12 @@ def main(cls):
                                 help="an ElasticSearch namespace",
                                 default=None)
             parser.add_argument("--indexer",
-                                help="whether this system does indexing",
-                                action='store_true',
-                                default=False)
+                                help="whether this server does indexing at all",
+                                choices=["true", "false"],
+                                default=None)
```

Comment:
Is None really what we want here? Will it pick up a default option elsewhere? Same question for `--index_server`, I think default should be `False` in that case.

Reply:
Yes, the problem is that we either have to have a rule that the command arguments override the environment variables or vice versa. The rule was (and I guess I could change this because it isn't clear it's right) that the arguments override the env variables, and so putting False here made the env variables not work. But anyway there is logic that picks up the env variables and defaults it. (This is all heavily unit tested but could still have gone by you as even then it's subtle.)

Another complexity in the design space is that env variables are strings and command arguments can be real datatypes. That's weird because when the env variables are merged, you also have to do various kinds of coercions to even get them on a level playing field, and THEN you have to flatten them again to strings to put them in an ini file and AGAIN recover them as strings later. I REALLY wish we were using a json structure to hold this data so we didn't have to do that second coerce-to-string.

Reply:
I'm leaving this alone for now.